### DISCUSSION TITLE
HEEELP! Spatie\browsershot not working. Always showing node_modules error.

### DISCUSSION THREAD

**Discussion by jamessoncardozo** [spatie/browsershot#722] [Bugs]
I'm facing a awful issue to make some screenshots with this library.  I'm tired. It does't matter what I can make. It showing a big error. Take a look.
Symfony\Component\Process\Exception\ProcessFailedException: The command &quot;node ^&quot;C:^\Program Files^\nodejs^\node.exe^&quot; ^&quot;^{^^&quot;url^^&quot;:^^&quot;https:^/^/google.com^^&quot;,^^&quot;action^^&quot;:^^&quot;screenshot^^&quot;,^^&quot;options^^&quot;:^{^^&quot;type^^&quot;:^^&quot;png^^&quot;,^^&quot;path^^&quot;:^^&quot;googlescreenshot.jpg^^&quot;,^^&quot;args^^&quot;:^[^],^^&quot;viewport^^&quot;:^{^^&quot;width^^&quot;:3840,^^&quot;height^^&quot;:2160^},^^&quot;landscape^^&quot;:true,^^&quot;waitUntil^^&quot;:^^&quot;networkidle0^^&quot;^}^}^&quot;&quot; failed.

Exit Code: 1(General error)

Working directory: C:\laravel\browsershot\public

Output:
================

Error Output:
================
&#039;node&#039; n�o � reconhecido como um comando interno
ou externo, um programa oper�vel ou um arquivo em lotes.
 in file C:\laravel\browsershot\vendor\spatie\browsershot\src\Browsershot.php on line 914
 
I have already reviewed the settings in vendor......\browsershot.php
I'm thinking that i'ts a problem with node and npm brinary path. I have tried specify the binary module, but still don't work.
The file "C:\laravel\browsershot\vendor\spatie\browsershot\src\Browsershot
.php" have this property. Can this be the problem? I'm using Windows 11 and Laragon.
 protected $includePath = '$PATH:/usr/local/bin:/opt/homebrew/bin';

![](https://cdn.discordapp.com/attachments/1091085575366381598/1091087799643545650/image.png)

**Comment by ahaneef29** [comment]
Try to install 16 or 17 version of peppeteer   "npm install peppeteer@17.1.3"

**Comment by rodolforjunior** [comment]
> I'm facing a awful issue to make some screenshots with this library. I'm tired. It does't matter what I can make. It showing a big error. Take a look. Symfony\Component\Process\Exception\ProcessFailedException: The command "node ^"C:^\Program Files^\nodejs^\node.exe^" ^"^{^^"url^^":^^"https:^/^/google.com^^",^^"action^^":^^"screenshot^^",^^"options^^":^{^^"type^^":^^"png^^",^^"path^^":^^"googlescreenshot.jpg^^",^^"args^^":^[^],^^"viewport^^":^{^^"width^^":3840,^^"height^^":2160^},^^"landscape^^":true,^^"waitUntil^^":^^"networkidle0^^"^}^}^"" failed.
> 
> Exit Code: 1(General error)
> 
> Working directory: C:\laravel\browsershot\public
> # Output:
> # Error Output:
> 
> 'node' n�o � reconhecido como um comando interno ou externo, um programa oper�vel ou um arquivo em lotes. in file C:\laravel\browsershot\vendor\spatie\browsershot\src\Browsershot.php on line 914
> 
> I have already reviewed the settings in vendor......\browsershot.php I'm thinking that i'ts a problem with node and npm brinary path. I have tried specify the binary module, but still don't work. The file "C:\laravel\browsershot\vendor\spatie\browsershot\src\Browsershot .php" have this property. Can this be the problem? I'm using Windows 11 and Laragon.  protected $includePath = '$PATH:/usr/local/bin:/opt/homebrew/bin';
> 
> ![](https://camo.githubusercontent.com/889e9daab881fb171fad607a0db4db1f45f8d0e0776ed3eba9bbc88a02469ff3/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f313039313038353537353336363338313539382f313039313038373739393634333534353635302f696d6167652e706e67)

Hello @jamessoncardozo,

Have you managed to find a fix for this issue?

I'm having the same trouble. Downgrading puppeteer did not work for me as suggested by the only response in this thread.

**Reply by jamessoncardozo** [reply]
I gave up on using spatie/browsershot. I'm using simplesoftwareio/simple-qrcode instead. It works independently of the environment, whether it's in production or development.

**Reply by rodolforjunior** [reply]
Thanks for the reply. I also gave up working with the library. I'll have a look at simple-qrcode.

Appreciate the input, cheers!

**Comment by jamessoncardozo** [comment]
Here is my Qrcode Generator. I dont know but this text editor is cutting a pice of my code.
![image](https://github.com/spatie/browsershot/assets/112659959/d5890092-2618-4458-ac7f-2b3bdc51c5e9)
`
<?php
namespace App\Traits;

use SimpleSoftwareIO\QrCode\Facades\QrCode;

trait MakeQrCodeTrait
{
  public function generateQrCode($user_name)
  {
![image](https://github.com/spatie/browsershot/assets/112659959/d5890092-2618-4458-ac7f-2b3bdc51c5e9)

    QrCode::format('png')
    ->size(399)
    ->margin(1)
    ->color(40,40,40)
    ->generate(env('APP_URL') . '/' . $user_name,'./img/qrcodes/'.$user_name.'.png');
  }
}
`
